penrose-web/src/engine/Evaluator.ts

```diff
@@ -252,44 +260,23 @@ export const evalExpr = (
 
     case "AFloat": {
       if (e.contents.tag === "Vary") {
+        console.error("expr", e, "trans", trans, "varyingVars", varyingVars);
         throw new Error("encountered an unsubstituted varying value");
```

Comment:
Seems like we are throwing lots of errors everywhere. Would be good to have a module just for error msgs and just call something like `throw UnsubstitutedVaryingError()` here.

Reply:
Sure, I'm happy to use those kinds of error messages in Evaluator if you want to set up an error message module (or send an example of how to do it).